lib/src/dialog/primary_dialog.dart

```diff
@@ -0,0 +1,207 @@
+  final String titleText;
+
+// The main content of the dialog box
+  final String contentText;
+
+// border radius of the dialog box [default: 5]
+  final double? dialogRadius;
+
+// color of the dialog box [default: white]
+  final Color? dialogBackgroundColor;
+
+// text style of the dialog box [default: size=18, bold, color=black]
+  final TextStyle? titleStyle;
+
+// text style of the dialog box [default: color=black54]
+  final TextStyle? contentStyle;
+
+// text on the primary button
+  final String primaryCta;
+
+// onTap functionality of the primary button
+  final Function()? primaryOnTap;
+
+// text on the secondary button
+  final String secondaryCta;
+
+// onTap functionality of the secondary button
+  final Function()? secondaryonTap;
+
+// background color of the primary button [default: black]
+  final Color? primaryCtaBackgroundColor;
+
+// foreground color of the primary button [default: white]
+  final Color? primaryCtaForegroundColor;
+
+// border color of the primary button [default: black]
+  final Color? primaryCtaBorderColor;
+
+// background color of the secondary button [default: white]
+  final Color? secondaryCtaBackgroundColor;
+
+// foreground color of the secondary button [default: black]
+  final Color? secondaryCtaForegroundColor;
+
+// border color of the secondary button [default: black54]
+  final Color? secondaryCtaBorderColor;
+
+  @override
+  State<MUIPrimaryDialog> createState() => _MUIPrimaryDialogState();
+}
+
+class _MUIPrimaryDialogState extends State<MUIPrimaryDialog> {
+  @override
+  Widget build(BuildContext context) {
+    // inkwell to have add the onTap functionality to the button
+    return InkWell(
+      onTap: () => showDialog(
+        context: context,
+        builder: (context) => AlertDialog(
+          // change the shape of the dialog box to bordered rect
+          shape: RoundedRectangleBorder(
+            borderRadius: BorderRadius.circular(
+              widget.dialogRadius ?? 5,
+            ),
+          ),
+          backgroundColor: widget.dialogBackgroundColor ?? Colors.white,
+          surfaceTintColor: widget.dialogBackgroundColor ?? Colors.white,
+
+          // title text of the dialog
+          title: Text(
+            widget.titleText,
+            style: widget.titleStyle ??
+                const TextStyle(
+                  fontWeight: FontWeight.w600,
+                  fontSize: 18,
+                ),
+          ),
+
+          // content text of the dialog
+          content: Text(
+            widget.contentText,
+            style:
+                widget.contentStyle ?? const TextStyle(color: Colors.black54),
+          ),
+
+          // action buttons of the dialog
+          actions: [
+            // secondary button
+            ElevatedButton(
+              style: ElevatedButton.styleFrom(
+                  padding: const EdgeInsets.symmetric(
+                    horizontal: 20,
+                    vertical: 20,
```

Comment:
Hey I would suggest to make these static padding values to dynamic like it should be multiplied with the `getScreenWidth(context)` * `some_double_value`

Reply:
Hey,
Doing that would not make it look standard and make it change the size based on the screen width, which I believe isn't a good idea.
We do have an open issue with the size that the button changes w.r.t the screen width. 
Not a good practice!

Reply:
Yeah I am thinking the same rn. Just busy with some other work. Currently I am merging this. Thank you for your valuable contribution and pointing out the issue ❤️.